### PR TITLE
[tests] Add linting to tests

### PR DESCRIPTION
**Pull request by gregnazario**
### Description
I had found some previous bugs in the tests via linting, as well as this should help keep our code maintainable in tests as well.

Many of the rules are ignored for one reason or another, but this gets rid of a lot of the violations.

### CODE REVIEW

File: tests/e2e/api/collection.test.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 // Copyright © Aptos Foundation
 // SPDX-License-Identifier: Apache-2.0
 
-import { Aptos, AptosConfig, Network } from "../../../src";
+// import { Aptos, AptosConfig, Network } from "../../../src";
```
Comment (heliuchuan) [inline tests/e2e/api/collection.test.ts:4]: intentional?

**Review by heliuchuan — APPROVED** [review]

File: tests/e2e/api/collection.test.ts (hunk 1 above)
Comment (gregnazario) [reply tests/e2e/api/collection.test.ts:4]: Yeah, failed the linter because it's unused imports

Comment (heliuchuan) [reply tests/e2e/api/collection.test.ts:4]: ahhh right i commented out the tests.  didn't have that context in the snippet view.

**Review by 0xmaayan — APPROVED** [review]
big change, lets merge it now before other PR coming to the queue lol

**Pull request merged**